File: src/main/java/ru/geekbrains/micecreator/models/complex/Accommodation.java
package ru.geekbrains.micecreator.models.complex;

import lombok.Data;
import lombok.NoArgsConstructor;
import ru.geekbrains.micecreator.models.basic.AccommodationType;
import ru.geekbrains.micecreator.models.basic.Room;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import java.math.BigDecimal;
import java.util.Date;

@Entity
@Table(name = "accommodations")
@NoArgsConstructor
@Data
public class Accommodation {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id_accommodations")
	private Integer id;

	@Column(name = "check_in_date")
	private Date checkInDate;

	@Column(name = "check_out_date")
	private Date checkOutDate;

	@Column(name = "pax")
	private Integer pax;

	@Column(name = "price")
	private BigDecimal price;

	@ManyToOne
	@JoinColumn(name = "tour_id")
	private Tour tour;

	@ManyToOne
	@JoinColumn(name = "room_id")
	private Room room;

	@ManyToOne
	@JoinColumn(name = "acc_type_id")
	private AccommodationType accType;

}

File: MCFrontend/src/app/service/complex/accommodation/accommodation.ts
import {ShortForm} from "../../basic/shortform";

export class Accommodation {

  constructor(public id: any,
              public checkInDate: Date,
              public checkOutDate: Date,
              public pax: number,
              public price: number,
              public nettoPrice: number,
              public tourId: number,
              public room: ShortForm,
              public accType: ShortForm,
              public hotel: ShortForm,
              public nights: number,
              public roomCount: number,
              public total: number,
              public nettoTotal: number,
              public creationDate: Date
  ) {
  }
}
